### PR TITLE
solution

### PR DESCRIPTION
**Pull request by abcdesign1**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',' || 'null') {
+    if (this.length === 0) {
+      return '';
+    }
```
Comment (stanislavkulakovskyi) [inline src/arrayMethodJoin.js:10]: this condition is unnecessary

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',' || 'null') {
```
Comment (stanislavkulakovskyi) [inline src/arrayMethodJoin.js:7]: why do you need this **|| 'null'** here?

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,8 +4,33 @@
...
+      return '';
+    }
+
+    let joined = '';
+
+    let newSeparator = separator;
+
+    if (separator === null) {
+      newSeparator = 'null';
+    }
```
Comment (stanislavkulakovskyi) [inline src/arrayMethodJoin.js:18]: this one is unnecessary as well

File: src/arrayMethodJoin.js [hunk 4]
```diff
@@ -4,8 +4,33 @@
...
+      if (i === this.length - 1) {
+        joined += this[i] || '';
+        break;
+      }
+
+      if (this[i] === undefined || this[i] === null) {
+        joined += '' + newSeparator;
+        continue;
+      }
+      joined += this[i] + newSeparator;
```
Comment (stanislavkulakovskyi) [inline src/arrayMethodJoin.js:30]: I guess it's better to write condition something like this:

      if (this[i] !== undefined && this[i] !== null) {
        joined += this[i];
      }

      if (i !== this.length - 1) {
        joined += separator;
      }
      
this way it looks more clear and compact

**Review by stanislavkulakovskyi — CHANGES_REQUESTED** [review]
good job overall, but there are some things that could be improved!

File: src/arrayMethodJoin.js (hunk 1 above)
Comment (vadym4che) [reply src/arrayMethodJoin.js:10]: fixed

File: src/arrayMethodJoin.js (hunk 2 above)
Comment (vadym4che) [reply src/arrayMethodJoin.js:7]: fixed

File: src/arrayMethodJoin.js (hunk 3 above)
Comment (vadym4che) [reply src/arrayMethodJoin.js:18]: fixed

**Review by stanislavkulakovskyi — APPROVED** [review]
Well done mate!

File: src/arrayMethodJoin.js [hunk 5]
```diff
@@ -4,8 +4,20 @@
...
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        joinedString += this[i];
+      }
+
+      if (i !== this.length - 1) {
```
Comment (OleksiiBlezniuk) [inline src/arrayMethodJoin.js:15]:
```suggestion
      if (i < this.length - 1) {
```

Comment (OleksiiBlezniuk) [reply src/arrayMethodJoin.js:15]: This way is more logical as for me

File: src/arrayMethodJoin.js [hunk 6]
```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
```
Comment (OleksiiBlezniuk) [inline src/arrayMethodJoin.js:8]: Ok in this context but better to avoid using data type in variables names

**Review by OleksiiBlezniuk — APPROVED** [review]

**Pull request closed without merging**